utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/swagger/SwaggerTyped.scala

```diff
@@ -171,23 +174,32 @@ object SwaggerArray {
 }
 
 object SwaggerObject {
+  private implicit val patternKeyDecoder: KeyDecoder[Pattern] = KeyDecoder[String].map(Pattern.compile)
+  private implicit val patternKeyEncoder: KeyEncoder[Pattern] = KeyEncoder[String].contramap(_.pattern())
+
   private[swagger] def apply(schema: Schema[Object], swaggerRefSchemas: SwaggerRefSchemas, usedRefs: Set[String]): SwaggerTyped = {
     val properties = Option(schema.getProperties).map(_.asScala.toMap.mapValuesNow(SwaggerTyped(_, swaggerRefSchemas, usedRefs)).toMap).getOrElse(Map())
 
     if (properties.isEmpty) {
       schema.getAdditionalProperties match {
-        case a: Schema[_] => SwaggerMap(Some(SwaggerTyped(a, swaggerRefSchemas, usedRefs)))
+        case a: Schema[_] if schema.getPatternProperties == null || schema.getPatternProperties.isEmpty => SwaggerMap(Some(SwaggerTyped(a, swaggerRefSchemas, usedRefs)))
         case b if b == false => new SwaggerObject(Map.empty, AdditionalPropertiesDisabled)
```

Comment:
here in `SwaggerObject`, why don't you handle `patternProperties` ?

Reply:
fixed + test case added